[Feat/crawler inflearn] 인프런 기본 베이스 생성

Junhyeock opened the pull request:
PR 요약
인프런 기본 베이스 생성

 상세 내용
- [x]  데이터 정규화
- [x] 강의 id를 딕셔너리에 저장하여 중복 제거
- [x] 병렬 처리
- [x] 비동기 구조  

기타 참고 사항

 PR Checklist
- [x] 커밋 메시지 컨벤션에 맞게 작성했습니다.

Yong-back committed:
✨ feat: recruitment_bookmarks serializer 작성

recruitment_bookmarks 의 serializer 작성하였습니다.

Yong-back committed:
✨ feat: recruitment_search_logs serializer 작성

recruitment_bookmarks serializer 작성 후 recruitment_search_logs serializer 작성 완료

Yong-back committed:
✨ feat: recruitment_bookmark view 작

recruitment_bookmark view.py 작성
성

UNGBI78 committed:
✨ [feat/withdraw] CKPS-8 탈퇴 상세 사유 빈칸 허용 안하게 변경

UNGBI78 committed:
✨ [feat/withdraw] CKPS-33 회원 탈퇴 기본 시리얼라이저 작성

Yong-back committed:
✨ feat: recruitment_bookmark 해당하는 view , test_code 작성

recruitment_bookmark 해당 views, test_code , serializer 작성 완료

UNGBI78 committed:
💡 [chore/whitdraw] CKPS-33 mypy 검사 타입 명시

UNGBI78 committed:
✨ [feat/withdraw] CKPS-34 탈퇴 기본 서비스 작성

UNGBI78 committed:
✅ [test/withdraw] CKPS-35 회원 탈퇴 models, services TestCode 작성

chdan-hub committed:
🐛 fix: withdraw_at에 따른 코드 수정

Yong-back committed:
bookmark serializer
-> bookmark_count source 제거 후 annotate 로 view 에서 들고올수있는 형식으로 view 수정

chdan-hub committed:
✨ feat: 유저 정보 및 상세 정보 조회 테스트 코드 추후 수정 필요

s4ngmin-9 committed:
✨ feat: drf-spectacular 요청/응답 스키마 분리를 위한 Request/Response Serializer 추가

- 요청 스키마 분리를 위한 Request Serializer 추가
- 공통 응답 스키마 분리를 위해 response_serializer.py 신규 생성

Yong-back committed:
💡 chore    : views ,serializer 코드 수

serializer 내에 view_count -> views_count 로 수정
views 코드 bookmark_count 에 해당하는 내용 수정 및
self.request.user -> request.user 로 수정
정

Commit:
✨ feat     : recruiter_application_view 작성

s4ngmin-9 committed:
✨ feat: MessageCreateRequestSerializer에 요청 데이터 유효성 검증 추가

리뷰해주신 내용 추가하였습니다

Junhyeock committed:
feat    : 기본 인프런 크롤러 작성
feat: 기본 인프런 크롤러 작성

2. 변경 또는 추가사항을 아래에 간략하게 작성하세요 ( 필수 )
병렬처리 로직 최적화 및 mypy 오류 해결

Junhyeock committed:
fix      : 필요없는 모듈 제거

romiii35 committed:
✨ feat : recruitment_serializer 작성

recruitment_serializer 작성

Yong-back committed:
💡 chore    : recruitment_bookmark view 수정

recruitment_bookmark view 수정

SWWS97 commented on apps/lectures/crawler/inflearn/inflearn_crawler.py:
```diff
@@ -0,0 +1,183 @@
+            "lang": "ko",
+        }
+
+        rating_text: Dict[int, str] = {
```
이부분 클래스 변수로 만들어서 아래에서 사용하는게 성능, 가독성 면에서 좋을거 같아요

Junhyeock replied:
넵 확인입니다

2joonkim commented on apps/lectures/crawler/inflearn/inflearn_crawler.py:
```diff
@@ -0,0 +1,183 @@
+                "thumbnail": quote(info.get("thumbnailUrl"), safe=":/?&"),
+                "url": f"https://www.inflearn.com/course/{info.get('slug')}",
+                "categories": [
+                    x.get("title") for x in info.get("metadata", {}).get("categories", []) if x.get("title")
```
<img width="308" height="209" alt="Image" src="https://github.com/user-attachments/assets/0de99768-7daa-43f9-b71a-7abfa3f51ab1" />

가격이 비어있는거 같습니다! 확인해주세요

2joonkim commented on apps/lectures/crawler/inflearn/inflearn_crawler.py:
```diff
@@ -0,0 +1,183 @@
+            3: "3_OUT_OF_5_STARS",
+            2: "2_OUT_OF_5_STARS",
+            1: "1_OUT_OF_5_STARS",
+        }
```
이 부분 int로 들어갑니다. rating text가 불필요한 거 같아요

2joonkim commented on apps/lectures/crawler/inflearn/inflearn_crawler.py:
```diff
@@ -0,0 +1,183 @@
+            data = res.json()
+            return cast(int, data.get("data", {}).get("totalPage", 1))
+        except Exception:
+            return 0
```
몇몇의 except Exception 부분들이 에러메세지 확인이 불가능하게 작성해 두셨습니다
이 부분 확인하셔서 logging 추가하시고 에러메시지 확인하게 바꿔보시는ㄱ ㅔ 디버깅하기 좋을 거 같아요

2joonkim commented on apps/lectures/crawler/inflearn/inflearn_crawler.py:
api 명세서랑 다르게 몇몇의 값들이 조금씩 뻐져있거나 다릅니다!
한번씩만 확인해주세요!!

Junhyeock replied:
네 확인하겠습니다

romiii35 committed:
💡 chore : recruitment_serializer 수정

recruitment_serializer 수정

romiii35 committed:
💡 chore : recruitment_serializer 수정

 recruitment_serializer 수정

romiii35 committed:
💡 chore : recruitment_serializer 오타 수정 및 안쓰는 import 정리

recruitment_serializer 오타 수정 및 안쓰는 import 정리

Commit:
💡 chore: 코드 전체적 수정

wjdtjddns98 commented on apps/lectures/crawler/inflearn/inflearn_crawler.py:
```diff
@@ -0,0 +1,183 @@
+    print(f"총 강의 수: {len(data)}개")
+    print(f"총 실행 시간: {end - start:.2f}초")
+    print(json.dumps(data[0], indent=4, ensure_ascii=False))
+
```
프린트문 로깅처리 해주세용

chdan-hub committed:
✨ feat: 유저 정보 수정 요청용 시리얼라이저 작성

chdan-hub committed:
✨ feat: 유저 정보 수정 뷰 코드 작성

TeaCat-Develop committed:
🐛 fix: 스터디그룹 모델에 db_column 추가

그룹멤버, 스터디강의 등의 필드명이 ForeignKey를 사용함에도 _id를 붙이며 몇몇 외래키 관련 필드명 참조 시 오류 발생.
mypy 참조 오류 등 혼동이 있으므로, 관련 모델 일부 속성 추가 및 마이그레이션
관련하여 오류 발생하는 부분 함께 수정

TeaCat-Develop committed:
✨ feat     : 스터디그룹 view, serializer, url 추가

TeaCat-Develop committed:
🎨 style: code formatting

UNGBI78 committed:
✨ [feat/withdraw] CKPS-32 회원 탈퇴 기본 views 작성

chdan-hub committed:
🐛 fix: 테스트 코드 mypy오류 수정

chdan-hub committed:
🐛 fix: black, isort, mypy로 인한 수정

UNGBI78 committed:
✨ [feat/withdraw] 회원 탈퇴 마이그레이션 0003 추가

UNGBI78 committed:
✨ [feat/withdraw] 회원 탈퇴 url 추가

UNGBI78 committed:
✨ [feat/withdraw] CKPS-35 회원 탈퇴 view TestCode 추가 & 각 TestCode 설명 주석 추가

UNGBI78 committed:
🐛 [fix/withdraw] CKPS-34 회원 탈퇴 service 로직 else -> atomic 교체 ( 안정성 )

Yong-back committed:
💡 chore    :  recruitment/urls 의 recrutiment_bookmarks.py -> recruitment_bookmark 오타 수정

recruitment/urls 의 recrutiment_bookmarks -> recruitment_bookmarks 로 파일 명 오타 수정

Yong-back committed:
💡 chore    : recrutiment -> recruitment 파일명 오타 수정

recrutiment -> recruitment 파일명 오타수정

chdan-hub committed:
🐛 fix: 필드값 변경 및 is_active로 적용

Commit:
♻️ refactor: recruiter_application_test_view 헬퍼 메서드 추가

Junhyeock committed:
fix : 전체 코드 구조 개선
fix: 전체 코드 구조 개선

2. 변경 또는 추가사항을 아래에 간략하게 작성하세요 ( 필수 )
- 로깅 추가 후 에러메시지 확인 가능
- 클래스 변수로 가독성 확보
- rating text 삭제
- 가격 추가 및 api 명세서 확인
- http2 설치 후 속도 13~20분에서 12초로 대폭 감소

s4ngmin-9 committed:
♻️ refactor: 리뷰 반영하여 리팩토링

UNGBI78 committed:
💡 [chore/withdraw] CKPS-8 탈퇴 모델에 withdrawn_at 필드 추가

UNGBI78 committed:
✨ [feat/withdraw] CKPS-8 탈퇴 모델 필드 추가에 따른 마이그레이션 생성

chdan-hub committed:
✨ feat: 유저 정보 삭제 Spec 뷰 코드 작성

UNGBI78 committed:
💡 [chore/withdraw] 공통 urls 파일에 회원 탈퇴용 url 추가 (Conflicts: config/urls.py)

Junhyeock committed:
fix : 피드백 개선
fix: 피드백 개선

2. 변경 또는 추가사항을 아래에 간략하게 작성하세요
- 로거마다 다른 파일명 사용
- 프로젝트 루트 기준위치 변경
- 로거 핸들러 중복체크
- 개발, 운영 환경에 따른 로그레벨 추가
- f-string 제거
- 매핑 개선
- id, name 타입 명시
- created_at 응답 제거
- 리뷰 id 명시 변경
- rating_map 제거
